### ISSUE TITLE
Generate ARB file from code

### ISSUE THREAD

**Issue by jaumard** [localizely/intl_utils#46]
I saw on intl_translation you were posting you can be used as replacement of that package. 
I'm wondering if this package can be used to generate ARB files from code like I'm doing with intl_translation, as it's the part missing for me. 
Thanks

**Comment by jaumard** [comment]
Searching for the equivalent of `pub run intl_translation:extract_to_arb --output-dir=target/directory my_program.dart more_of_my_program.dart`

**Comment by aleksakrstic** [comment]
Hi @jaumard 
Yes, `intl_utils` is a replacement of `intl_translation` package and has the same generator foundation.
The main purpose is to generate Dart boilerplate code for localization from ARB files and get rid of manual work.

Since ARB files are the main source of truth for string keys, there is no need to generate them from Dart code additionally. This proved as more productive and simplified approach.

May I ask why you still want to write Dart code for l10n manually and generate ARB file from it, maybe I can suggest alternative solution..?

**Comment by jaumard** [comment]
> Since ARB files are the main source of truth for string keys,

Not for me :) my dart class is the main source of truth

Because I don't want to deal with ARB files :) much faster and safer to me to write a dart field in my class than just edit manually a json file... Also writing directly my dart code prevent me to regenerate anything until I totally finish my feature. 

My flow is like this: 
- writing my feature and add intl fields into my class
- generate ARB file once I'm done with the feature
- send it somewhere to be translated
- downloaded back once translated
- generate messgage_all files from ARB

To me that's the more productive way to deal with translations. Never had issues with that until  `intl_translation` look not maintained anymore...  

You can't really say to me that 
```
"airtimeBalanceUpdate": "{howMany,plural, =1{Last update {howMany} minute ago}other{Last update {howMany} minutes ago}}",
  "@airtimeBalanceUpdate": {
    "description": "Airtime balance screen",
    "type": "text",
    "placeholders": {
      "howMany": {}
    }
  },
```
Is better and safer than:
```
String airtimeBalanceUpdate(int howMany) => Intl.plural(howMany, one: 'Last update $howMany minute ago', other: 'Last update $howMany minutes ago', name: 'airtimeBalanceUpdate', args: [howMany], desc: 'Airtime balance screen');
```
If you only remember first the syntax to put in JSON lol 

I can understand for small project or key values translations it might not bother anyone to deal with ARB. But my project contains 1000 sentences so far and growing because we're white labeled (so also need to merge some translation from the base flavor) now and with some "complexe" plural and other stuff really well managed by intl, I don't know why I would want to loose that for writing JSON

**Comment by aleksakrstic** [comment]
If you use this `intl_utils` package or its IDE plugins for [VSCode](https://marketplace.visualstudio.com/items?itemName=localizely.flutter-intl) or [AndroidStudio](https://plugins.jetbrains.com/plugin/13666-flutter-intl) this is the minimal you need to write for your string key:

    "airtimeBalanceUpdate": "{howMany,plural, =1{Last update {howMany} minute ago}other{Last update {howMany} minutes ago}}"

Meta data `@airtimeBalanceUpdate` is optional.
If the [ICU string](https://devpal.co/icu-message-editor/) can not be parsed, you will see the error in CLI output or a comment instead of string getter in generated Dart class. So it is not something that breaks at runtime if you are afriad of that.
I am not sure if it gets harder this way if the project gets 1000 strings.
Maybe you can give it a try and share the insight from your use case?

**Comment by jaumard** [comment]
Even if I have to write only this: `"airtimeBalanceUpdate": "{howMany,plural, =1{Last update {howMany} minute ago}other{Last update {howMany} minutes ago}}"`  that super error prone and I'll never remember the syntax... So much waste of time where we have something in pure dart doing well the job... 

`@airtimeBalanceUpdate` is indeed optional, but not really if you want to offer context to your translators... and once again that come for free using the dart intl syntax. 

> Maybe you can give it a try and share the insight from your use case?

Yeah just tried... And I can't use it, you support only one ARB file where I have two (one for generic messages and another only for errors). Any way to support multi files ?

**Comment by Fintasys** [comment]
I agree with @jaumard . I want to add intl to my existing app and I would like to avoid to generate strings from < 100 classes.  `extract_to_arb` would come super handy for that reason and save lot of time. Additional supporting multiple files / folder would help a lot.

**Comment by aleksakrstic** [comment]
@jaumard Using multiple ARB files is blocked by internals of Intl package which is used under the hood.

@Fintasys Would ARB-first approach work in your case? 
Defining strings in ARB files firts, and using them in Dart code from auto-enerated l10n classes?

**Comment by jaumard** [comment]
@aleksakrstic I'm currently using multi arb in my project ^^ so that's possible.

**Comment by aleksakrstic** [comment]
@jaumard Do you execute `intl_translation:generate_from_arb` for each ARB file separately or?
Do you have only one Localization Delegate class?

**Comment by jaumard** [comment]
Yes I have two classes to really separate the meaning my translations and run the command for each of them.

**Comment by cyrax111** [comment]
`intl_utils` generates files for flutter only
